src/plugins/intel_gpu/src/kernel_selector/kernels/lstm/lstm_seq_kernel_base.cpp

```diff
@@ -0,0 +1,89 @@
+        MakeJitConstant("GEMM_OFFSET_Z", params.GetOffsetIndexZ() * size),
+    });
+    jit.AddConstants({MakeJitConstant("BATCH_SIZE", params.inputs[1].Batch().v)});
+    jit.AddConstants({MakeJitConstant("MAX_SEQ_LENGTH", params.inputs[0].Feature().v)});
+    jit.AddConstants({MakeJitConstant("INPUT_SIZE", params.inputs[0].Y().v)});
+    jit.AddConstants({MakeJitConstant("HIDDEN_SIZE", params.inputs[1].Y().v)});
+    auto ftype = params.inputs[0].GetDType();
+    // if ReLU activation present, we have to reset accumulator type for the kernel to FP32
+    // to avoid possible overflows on FP16, since ReLU doesn't limit upper border of its result
+    for (size_t i = 0; i < params.activations.size(); i++) {
+        if (params.activations[i].function == ActivationFunction::RELU) {
+            ftype = Datatype::F32;
+            break;
+        }
+    }
+    jit.Merge(MakeTypeJitConstants(ftype, "ACCUMULATOR"));
+
+    static const std::vector<std::string> asuffixes = {"_F", "_G", "_H", "_CLIP"};
+    for (size_t i = 0; i < params.activations.size(); i++) {
+        std::vector<base_activation_params> aparams = { params.activations[i] };
+        jit.Merge(MakeActivationJitConstants(aparams, params.inputs[0].GetDType(), asuffixes[i]));
+    }
+
+    if (params.clip <= 0) {
+        jit.AddConstants({
+                MakeJitConstant("ACTIVATION_PARAMS_CLIP", ""),
+                MakeJitConstant("ACTIVATION_CLIP(x, p)", "(x)"),
+            });
+    }
+
+    return jit;
+}
+
+KernelsData LSTMSeqKernelBase::GetCommonKernelsData(const Params& params) const {
+    if (!Validate(params)) {
+        return {};
+    }
+
+    const lstm_seq_params& orgParams = static_cast<const lstm_seq_params&>(params);
+
+    KernelData kd = KernelData::Default<lstm_seq_params>(params, 1);
+
+    auto out =  orgParams.outputs[0];
+
+    auto& kernel = kd.kernels[0];
+    kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 0});
+    kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 1});
+    kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 2});
+    kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 3});
+    kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 4});
+    kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 5});
+    kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 6});
+    kernel.params.arguments.push_back({ArgumentDescriptor::Types::OUTPUT, 0});
+    kernel.params.arguments.push_back({ArgumentDescriptor::Types::OUTPUT, 1});
+    kernel.params.arguments.push_back({ArgumentDescriptor::Types::OUTPUT, 2});
+    auto cldnnJit = GetJitConstants(orgParams);
+    auto entryPoint = GetEntryPoint(kernelName, orgParams.layerID, params);
+    auto jit = CreateJit(kernelName, cldnnJit, entryPoint);
+
+    kernel.params.workGroups.global = {out.X().v, out.Batch().v, 1};
+    kernel.params.workGroups.local = {out.X().v, 1, 1};
```

Comment:
I think this is not safe as X may be larger than allowed LWS

Reply:
updated